base/src/AudioToTextXForm.cpp

```diff
@@ -0,0 +1,180 @@
+}
+
+AudioToTextXForm::~AudioToTextXForm() {}
+
+bool AudioToTextXForm::validateInputPins()
+{
+	if (getNumberOfInputPins() != 1)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins size is expected to be 1. Actual<" << getNumberOfInputPins() << ">";
+		return false;
+	}
+
+	framemetadata_sp metadata = getFirstInputMetadata();
+
+	FrameMetadata::FrameType frameType = metadata->getFrameType();
+	if (frameType != FrameMetadata::AUDIO)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateInputPins input frameType is expected to be Audio. Actual<" << frameType << ">";
+		return false;
+	}
+
+	return true;
+}
+
+bool AudioToTextXForm::validateOutputPins()
+{
+	if (getNumberOfOutputPins() != 1)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins size is expected to be 1. Actual<" << getNumberOfOutputPins() << ">";
+		return false;
+	}
+
+	framemetadata_sp metadata = getFirstOutputMetadata();
+	FrameMetadata::FrameType frameType = metadata->getFrameType();
+	if (frameType != FrameMetadata::TEXT)
+	{
+		LOG_ERROR << "<" << getId() << ">::validateOutputPins input frameType is expected to be TEXT. Actual<" << frameType << ">";
+		return false;
+	}
+
+	return true;
+}
+
+void AudioToTextXForm::addInputPin(framemetadata_sp& metadata, string& pinId)
+{
+	Module::addInputPin(metadata, pinId);
+	mDetail->mOutputMetadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::TEXT));
+	mDetail->mOutputMetadata->copyHint(*metadata.get());
+	mDetail->mOutputPinId = addOutputPin(mDetail->mOutputMetadata);
+}
+
+bool AudioToTextXForm::init()
+{
+	//intialize model
+	auto samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_GREEDY;
+	switch (mDetail->mProps.samplingStrategy)
+	{
+		case AudioToTextXFormProps::DecoderSamplingStrategy::GREEDY:
+			samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_GREEDY;
+			break;
+		case AudioToTextXFormProps::DecoderSamplingStrategy::BEAM_SEARCH:
+			samplingStrategy = whisper_sampling_strategy::WHISPER_SAMPLING_BEAM_SEARCH;
+			break;
+		default:
+			throw AIPException(AIP_FATAL, "Unknown Sampling Strategy");
+	}
+	mDetail->mWhisperFullParams = whisper_full_default_params(samplingStrategy);
+	mDetail->mWhisperContextParams = whisper_context_default_params();
+	mDetail->mWhisperContext = whisper_init_from_file_with_params(mDetail->mProps.modelPath.c_str(), mDetail->mWhisperContextParams);
+	return Module::init();
+}
+
+bool AudioToTextXForm::term()
+{
+	whisper_free_context_params(&mDetail->mWhisperContextParams);
+	whisper_free_params(&mDetail->mWhisperFullParams);
+	whisper_free(mDetail->mWhisperContext);
+	return Module::term();
+}
+
+bool AudioToTextXForm::process(frame_container& frames)
+{
+	auto frame = frames.begin()->second;
+	sf::Int16* constFloatPointer = static_cast<sf::Int16*>(frame->data());
+	int numberOfSamples = frame->size() / 2;
+	for (int index = 0; index < numberOfSamples; index++) {
+		mDetail->inputAudioBuffer.push_back((float)constFloatPointer[index]/ 32768.0f);
```

Comment:
This conversion may be a costly implementation.  May be one option is to go ahead as-is with technical debt and later use [IPP](https://www.intel.com/content/www/us/en/developer/articles/tool/oneapi-standalone-components.html#ipp). They have made it a freeware with a simplified license so there is no harm in using it along with our software, it wont work on Jetsons but we can use npp there

Reply:
https://github.com/Apra-Labs/ApraPipes/issues/326 I created a TDebt issue